app/controllers/settings/two_factor_authentication/webauthn_credentials_controller.rb

```diff
@@ -3,9 +3,9 @@
 module Settings
   module TwoFactorAuthentication
     class WebauthnCredentialsController < BaseController
-      layout 'admin'
+      skip_before_action :require_functional!
 
-      before_action :authenticate_user!
+      before_action :require_not_suspended!
```

Comment:
As I said in my previous comment, I think it's probably fine to allow changing 2FA settings for suspended users.